src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -53,57 +53,109 @@ public EditCommand parse(String args) throws ParseException {
 
         EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
 
-        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
-            editPersonDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
-        }
-        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
-            editPersonDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
-        }
-        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
-            editPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+        parseNameIfAny(argMultimap, editPersonDescriptor);
+
+        parsePhoneIfAny(argMultimap, editPersonDescriptor);
+
+        parseEmailIfAny(argMultimap, editPersonDescriptor);
+
+        parseRoleIfAny(argMultimap, editPersonDescriptor);
+
+        parseEmploymentTypeIfAny(argMultimap, editPersonDescriptor);
+
+        parseExpectedSalaryIfAny(argMultimap, editPersonDescriptor);
+
+        parseLevelOfEducationIfAny(argMultimap, editPersonDescriptor);
+
+        parseExperienceIfAny(argMultimap, editPersonDescriptor);
+
+        parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
+
+        parseInterviewIfAny(argMultimap, editPersonDescriptor);
+
+        parseNotesIfAny(argMultimap, editPersonDescriptor);
+
+        if (!editPersonDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
         }
 
-        if (argMultimap.getValue(PREFIX_ROLE).isPresent()) {
-            editPersonDescriptor.setRole(ParserUtil.parseRole(argMultimap.getValue(PREFIX_ROLE).get()));
+        return new EditCommand(index, editPersonDescriptor);
+    }
+
+    private void parseNotesIfAny(ArgumentMultimap argMultimap, EditPersonDescriptor editPersonDescriptor)
```

Comment:
Maybe can add javadoc for these new methods :)

Reply:
Thanks for the suggestion. However, as the method name is self-explanatory and the method is private, I don't think the purpose of this method is complex or confusing enough to warrant writing javadocs.